torch_xla/csrc/ops/index_put.cpp

```diff
@@ -29,8 +29,8 @@ NodePtr IndexPut::Clone(OpList operands) const {
 }
 
 XlaOpVector IndexPut::Lower(LoweringContext* loctx) const {
-  auto add_scatter_combiner = [](const xla::XlaOp& x,
-                                 const xla::XlaOp& y) -> xla::XlaOp {
+  std::function<xla::XlaOp(xla::XlaOp, xla::XlaOp)> add_scatter_combiner =
```

Comment:
Why?
We use `auto` everywhere for this (embracing the templated-result-can-be-auto rule).

Reply:
With `auto` this would be a lambda. Do you suggest using `std::function<xla::XlaOp(xla::XlaOp, xla::XlaOp)>(...)` on the right hand side instead?

Reply:
No, leave it alone. It get's converted to `std::function` anyway (need to be passed to a function taking `std::function`).